File: separate_compilation/Haskell/Algorithm.hs
module Algorithm(g) 

where

f :: [Integer] -> [Integer] -> Integer -> (Integer,Integer) -> (Integer, Integer)
f [] _ d p = p
f _ [] d p = p
f (x:xs) (y:ys) d p
					| (abs(x - y) < d) && (x < y) = f xs (y:ys) (abs (x - y)) (x,y)
					| (abs(x - y) < d)            = f (x:xs) ys (abs (x - y)) (x,y)
					| x < y                       = f xs (y:ys) d p
					| otherwise                   = f (x:xs) ys d p






g :: [Integer] -> [Integer] -> (Integer, Integer)
g xs ys = f xs ys 99999 (0,0)

File: separate_compilation/Java/Algorithm.java
public class Algorithm{
	public Pair algorithm(int a[], int b[]){
		int i = 0;
		int j = 0;
		int n1 = 0;
		int n2 = 0;
		int difference = Integer.MAX_VALUE;
		
		while(i < a.length && j < b.length){
			if(Math.abs(a[i] - b[j]) < difference){
				difference = Math.abs(a[i] - b[j]);
				n1 = a[i];
				n2 = b[j];
			}
			
			if(a[i] < b[j]){
				i++;
			}
			else{
				j++;
			}
		}
		
		return new Pair(n1, n2);
	}
}